src/app/core/learning-object-module/search/search.service.ts

```diff
@@ -1,9 +1,100 @@
 import { Injectable } from '@angular/core';
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { LearningObject } from '@entity';
+import { catchError } from 'rxjs/operators';
+import { Subject, throwError } from 'rxjs';
+import { Query } from 'app/interfaces/query';
+import * as querystring from 'querystring';
+import { SEARCH_ROUTES } from 'app/core/learning-object-module/search/search.routes';
 
 @Injectable({
-  providedIn: 'root'
+  providedIn: 'root',
 })
 export class SearchService {
+  needsChange$: Subject<void> = new Subject();
 
-  constructor() { }
+  constructor(private http: HttpClient) {}
+
+  /**
+   * Fetches Array of Learning Objects
+   *
+   * @returns {Promise<LearningObject[]>}
+   * @memberof LearningObjectService
+   */
+  getPublicLearningObjects(
+    query?: Query,
+  ): Promise<{ learningObjects: LearningObject[]; total: number }> {
+    let route = '';
+    if (query) {
+      const queryClone = Object.assign({}, query);
+      if (
+        queryClone.standardOutcomes &&
+        queryClone.standardOutcomes.length &&
+        typeof queryClone.standardOutcomes[0] !== 'string'
+      ) {
+        queryClone.standardOutcomes = (
+          queryClone.standardOutcomes as string[]
+        ).map((o) => o['id']);
+      }
+      const queryString = new URLSearchParams(queryClone).toString();
+      route =
+        SEARCH_ROUTES.GET_PUBLIC_LEARNING_OBJECTS_WITH_FILTER(queryString);
```

Comment:
can we rename the search route to match the name we use in the backend? `searchLearningObjects` and then condense these functions into one?